Modules/Devices/test/Devices.Domain.Tests/Identities/StartDeletionProcessAsOwnerTests.cs

```diff
@@ -42,17 +43,36 @@ public void Start_deletion_process()
         auditLogEntry.NewStatus.Should().Be(DeletionProcessStatus.Approved);
     }
 
+    [Fact]
+    public void Throws_when_device_not_owned_by_identity()
+    {
+        // Arrange
+        SystemTime.Set(DateTime.Parse("2020-01-01"));
+        var identity = TestDataGenerator.CreateIdentity();
+        var device = new Device(identity);
+
+        identity.Devices.Add(device);
+        identity.StartDeletionProcessAsOwner(device.Id);
+
+        // Act
+        var acting = () => identity.CancelDeletionProcess(identity.DeletionProcesses[0].Id, DeviceId.Parse("DVC"));
+
+        // Assert
+        acting.Should().Throw<DomainException>().Which.Code.Should().Be("error.platform.recordNotFound");
```

Comment:
You should additionally assert that the message contains the name of the entity that wasn't found.

Reply:
Now you act twice. Look into other occurrences of this "pattern" or ask @NikolaVetnic